service/backend/pchain/indexer/parser.go

```diff
@@ -0,0 +1,340 @@
+package indexer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/genesis"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+
+	pBlocks "github.com/ava-labs/avalanchego/vms/platformvm/blocks"
+	pGenesis "github.com/ava-labs/avalanchego/vms/platformvm/genesis"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	proposerBlk "github.com/ava-labs/avalanchego/vms/proposervm/block"
+
+	"github.com/ava-labs/avalanche-rosetta/client"
+	"github.com/ava-labs/avalanche-rosetta/mapper"
+)
+
+var errParserUninitialized = errors.New("uninitialized parser")
+
+const genesisTimestamp = 1599696000
```

Comment:
Can we use a date and convert it to its timestamp representation instead to make this more verbose

Reply:
Sure